execution-engine/contracts/system/client-api-proxy/src/client_api/mod.rs

```diff
@@ -76,13 +88,21 @@ impl Api {
 
                 runtime::call_contract(
                     pos_ref,
-                    (internal_method_names::BOND, amount, bonding_purse),
+                    (method_names::internal::BOND, amount, bonding_purse),
                 )
             }
             Self::Unbond(amount) => {
                 let pos_ref = system::get_proof_of_stake();
                 let amount: Option<U512> = amount.map(Into::into);
-                runtime::call_contract(pos_ref, (internal_method_names::UNBOND, amount))
+                runtime::call_contract(pos_ref, (method_names::internal::UNBOND, amount))
+            }
+            Self::StandardPayment(amount) => {
+                let pos_ref = system::get_proof_of_stake();
+                let main_purse = account::get_main_purse();
+                let payment_purse: PurseId =
```

Comment:
When calling from outside, an error of `Exit Code: 15 (Transfer)` occurs. I think it started from that part. Is there a problem you guessed?

Reply:
Could you explain how do you make a deploy so I can reproduce your problem?

Reply:
```
[]*consensus.Deploy_Arg{
		&consensus.Deploy_Arg{
			Name: "method",
			Value: &consensus.Deploy_Arg_Value{
				Value: &consensus.Deploy_Arg_Value_StringValue{
					StringValue: "standard_payment"}}},
		&consensus.Deploy_Arg{
			Name: "fee",
			Value: &consensus.Deploy_Arg_Value{
				Value: &consensus.Deploy_Arg_Value_BigInt{
					BigInt: &state.BigInt{
						Value:    "10000000",
						BitWidth: 512,
					}}}}}
```

like that.. using go util.. I think it's argument is same to bond.. But bond work is good.

Reply:
I do not know why Bond successes exactly(its `amount` arg is also u64), but you need to exactly match the type of the arguments(for standard_payement, the `amount` is u32 type).

Reply:
Ohohohohohohohoh it works!
But shouldn't we use the bigint U512?

Reply:
It would be great to discuss the type of the `amount` in API(bond, unbond...).

I thought it would be okay with `u64`, `u32` for commonly used API. Most users usually use the value under the range of those types, and then they can request with less verbose JSON conveniently. Furthermore, if they need to transfer or bond or pay a huge amount, they can do through their own contract.

But it is welcome to discuss this~!

Reply:
If we use 18 decimal places, 1HDAC will be 1000000000000000000, and it will be overflow after 1000HDAC.

Reply:
How about postponing to decide this type until we determine the gas price and denomination?